File: viewutils/src/main/java/io/vithor/yamvpframework/viewutils/IntentHelper.java
package io.vithor.yamvpframework.viewutils;

import android.content.ContentResolver;
import android.content.ContentValues;
import android.content.Context;
import android.content.Intent;
import android.content.pm.PackageManager;
import android.content.pm.ResolveInfo;
import android.net.Uri;
import android.provider.MediaStore;
import android.support.v4.content.IntentCompat;

import java.util.List;

/**
 * Created by Vithorio Polten on 1/4/16.
 */
public class IntentHelper {

    public static Intent getCameraIntent(ContentValues values, Uri imageUri, ContentResolver contentResolver) {
        Intent cameraIntent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
        cameraIntent.putExtra(MediaStore.EXTRA_OUTPUT, imageUri);
        return cameraIntent;
    }

    public static Intent getGalleryIntent() {
        Intent i = new Intent(Intent.ACTION_PICK, android.provider.MediaStore.Images.Media.EXTERNAL_CONTENT_URI);
        i.setType("image/*");
        return i;
    }

    public static boolean isIntentSafe(Context context, Intent intent) {
        PackageManager packageManager = context.getPackageManager();
        List<ResolveInfo> activities = packageManager.queryIntentActivities(intent, 0);
        return activities.size() > 0;
    }

    public static Intent newClearTask(Context context, Class<?> cls) {
        Intent intent = new Intent(context, cls);
        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | IntentCompat.FLAG_ACTIVITY_CLEAR_TASK);
        return intent;
    }
}

File: ui/src/main/java/io/vithor/yamvpframework/ui/IntentHelper.kt
@file:Suppress("UNUSED_PARAMETER", "unused")

package io.vithor.yamvpframework.ui

import android.content.ContentResolver
import android.content.ContentValues
import android.content.Context
import android.content.Intent
import android.net.Uri
import android.provider.MediaStore
import android.support.v4.content.IntentCompat

/**
 * Created by Vithorio Polten on 1/4/16.
 */
object IntentHelper {

    fun getCameraIntent(values: ContentValues, imageUri: Uri, contentResolver: ContentResolver): Intent {
        return Intent(MediaStore.ACTION_IMAGE_CAPTURE).apply {
            putExtra(MediaStore.EXTRA_OUTPUT, imageUri)
        }
    }

    val galleryIntent: Intent
        get() {
            val i = Intent(Intent.ACTION_PICK, android.provider.MediaStore.Images.Media.EXTERNAL_CONTENT_URI)
            i.type = "image/*"
            return i
        }

    fun isIntentSafe(context: Context, intent: Intent): Boolean {
        val packageManager = context.packageManager
        val activities = packageManager.queryIntentActivities(intent, 0)
        return activities.size > 0
    }

    fun newClearTask(context: Context, cls: Class<*>): Intent {
        return Intent(context, cls).apply {
            flags = Intent.FLAG_ACTIVITY_NEW_TASK or IntentCompat.FLAG_ACTIVITY_CLEAR_TASK
        }
    }
}
